src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -120,10 +120,8 @@ def build_alibi_tensor(attention_mask: torch.Tensor, n_head: int, dtype, device)
     # This is more or less identical to T5's relative position bias:
     # https://github.com/huggingface/transformers/blob/f681437203baa7671de3174b0fa583c349d9d5e1/src/transformers/models/t5/modeling_t5.py#L527
     # batch_size = 1, n_head = n_head, query_length
-
-    arange_tensor = (attention_mask.cumsum(-1)[:, None, :].to(device) - 1) * attention_mask[:, None]
-    alibi = slopes.unsqueeze(-1) * arange_tensor
-    alibi = alibi * attention_mask[:, None]
+    arange_tensor = ((attention_mask.cumsum(-1) - 1) * attention_mask)[:, None, :]
+    alibi = slopes[..., None] * arange_tensor
```

Comment:
This is not cast to `device` anymore.

Reply:
Yes but is there a need to, this should `device` should always be `attention_mask.device`

Reply:
Then  remove the device argument in this function then, since it's not used anywhere anymore.

Reply:
Yep should be done.